Refactor readme.md

chiukapoor opened the pull request:
### Updates
* Add `kind` cluster to the example
* Update the commands in the Getting Started section to run locally
* Unify the gunzip and tar command
* Updated WordPress example yaml location
* Add plugin, venv, and provider-kubeconfig.log in .gitignore
* Remove `parse-api-server-url.sh` file and get apiserver URL using kubeconfig

devdattakulkarni commented on README.md:
```diff
@@ -25,111 +26,115 @@ KubePlus takes an application Helm chart and wraps it in a Kubernetes API (CRD).
+Let’s look at an example of creating a multi-instance WordPress Service using KubePlus. The WordPress service provider goes through the following steps towards this on their cluster:
 
-## Getting Started 
+1. Clone the repository locally
```
Good addition. I suggest we also include the clone command with shallow depth:
git clone --depth 1 https://github.com/cloud-ark/kubeplus.git

This is the guidance that we have been providing in the examples since the size of the repo is big. 
Regarding the size - I will add thoughts to the issue.

devdattakulkarni commented on README.md:
````diff
@@ -25,111 +26,115 @@ KubePlus takes an application Helm chart and wraps it in a Kubernetes API (CRD).
-  export PATH=$KUBEPLUS_HOME/plugins:$PATH
-  kubectl kubeplus commands
-```
+   `kind create cluster`
````
Nice!

We had not tested with kind yet. Did everything work okay with kind cluster?

chiukapoor replied:
Yes, I tested it on `kind` cluster and everything was working.

devdattakulkarni commented on README.md:
```diff
@@ -25,111 +26,115 @@ KubePlus takes an application Helm chart and wraps it in a Kubernetes API (CRD).
    pip3 install -r requirements.txt
    apiserver=`./parse-api-server-url.sh`
    python3 provider-kubeconfig.py -s $apiserver create $KUBEPLUS_NS
+   deactivate
```
👍

devdattakulkarni commented on README.md:
```diff
@@ -25,111 +26,115 @@ KubePlus takes an application Helm chart and wraps it in a Kubernetes API (CRD).
    
-   The WordPress Helm chart can be specified as a [public url](./examples/multitenancy/wordpress/wordpress-service-composition.yaml#L14) or can be [available locally](./examples/multitenancy/wordpress/wordpress-service-composition-localchart.yaml#L14).
+   *The WordPress Helm chart can be specified as a [public url](./examples/multitenancy/wordpress/wordpress-service-composition.yaml#L14) or can be [available locally](./examples/multitenancy/application-hosting/wordpress/wordpress-service-composition.yaml).*
 
```
Locally available chart is available in wordpress-service-composition-localchart.yaml.

devdattakulkarni commented on README.md:
````diff
@@ -25,111 +26,115 @@ KubePlus takes an application Helm chart and wraps it in a Kubernetes API (CRD).
 
    ```
-   kubectl create -f  https://raw.githubusercontent.com/cloud-ark/kubeplus/master/examples/multitenancy/wordpress/wordpress-service-composition.yaml --kubeconfig=kubeplus-saas-provider.json
+   kubectl create -f examples/multitenancy/application-hosting/wordpress/wordpress-service-composition.yaml --kubeconfig=kubeplus-saas-provider.json
````
Some background on why we have absolute paths here.
We have gone back and forth on whether to list the relative paths or absolute paths. For relative paths, the repo has to be cloned. And given the size of the repo, we felt that it might become a blocker for someone to try the example. That was the reason we did not have the "clone the repo" as the first step. Anyone who wants to try, should be able to do so without having to first clone the repo. Of course, once the repo size is fixed cloning will not be an issue. And as a temporary solution, shallow clone also will be helpful.
What do you think?

devdattakulkarni commented on .gitignore:
```diff
@@ -1,4 +1,7 @@
 *~
+plugins
+venv
+provider-kubeconfig.log
```
👍

chiukapoor commented:
As discussed on Slack, I have updated the readme and removed the dependency to clone the repo till we fix the [issue](https://github.com/cloud-ark/kubeplus/issues/1281)

devdattakulkarni commented:
Looks good!
